repositories/src/test/scala/cromwell/pipeline/datastorage/dao/repository/UserRepositoryTestImp.scala

```diff
@@ -0,0 +1,59 @@
+package cromwell.pipeline.datastorage.dao.repository
+import cromwell.pipeline.datastorage.dto.User
+import cromwell.pipeline.model.wrapper.{ UserEmail, UserId }
+
+import scala.collection.mutable
+import scala.concurrent.Future
+
+class UserRepositoryTestImp extends UserRepository {
+
+  private val users: mutable.Map[UserId, User] = mutable.Map.empty
+
+  def getUserById(userId: UserId): Future[Option[User]] =
+    Future.successful(users.get(userId))
+
+  def getUserByEmail(email: UserEmail): Future[Option[User]] =
+    Future.successful(users.values.find(_.email == email))
+
+  def getUsersByEmail(emailPattern: String): Future[Seq[User]] =
+    Future.successful(users.values.filter(_.email.unwrap.contains(emailPattern)).toSeq)
+
+  def addUser(user: User): Future[UserId] = {
+    users += (user.userId -> user)
+    Future.successful(user.userId)
+  }
+
+  def deactivateUserByEmail(email: UserEmail): Future[Int] =
+    deactivateUser[UserEmail](email, user => user.email, (x, y) => x == y)
+
+  def deactivateUserById(userId: UserId): Future[Int] =
+    deactivateUser[UserId](userId, user => user.userId, (x, y) => x == y)
+
+  private def deactivateUser[A](userParam: A, toUserField: User => A, predicate: (A, A) => Boolean): Future[Int] = {
```

Comment:
Maybe that predicate parameter name could be more informative

Reply:
could you suggest some variants?

Reply:
isUserParamMatches, areUserParamsEquals. 
If I got it right, userParam and userField are the same entities. Maybe it will make sense to name them accordingly. Like 
(userParam: A, toUserParam: User => A, areUserParamsEquals: (A, A) => Boolean) or
(userField: A, toUserField: User => A, areUserFieldsEquals: (A, A) => Boolean) 

Btw, do we need that predicate parameter since it is the same in both deactivateUserBy methods? Cant we just replace it's call with " userParam == toUserField(user) "

Reply:
> Btw, do we need that predicate parameter since it is the same in both deactivateUserBy methods? Cant we just replace it's call with " userParam == toUserField(user) "

@Nikolai-Isaev  thanks for this idea. I'll do so.

Reply:
@Nikolai-Isaev done